synapse/federation/federation_client.py

```diff
@@ -992,3 +992,36 @@ def forward_third_party_invite(self, destinations, room_id, event_dict):
                 )
 
         raise RuntimeError("Failed to send to any server.")
+
+    @defer.inlineCallbacks
+    def get_room_complexity(self, destination, room_id):
+        """
+        Fetch the complexity of a remote room from another server.
+
+        Args:
+            destination (str): The remote server
+            room_id (str): The room ID to ask about.
+
+        Returns:
+            Deferred[dict] or Deferred[None]: Dict contains the complexity
+            metric versions, while None means we could not fetch the complexity.
+        """
+        try:
+            complexity = yield self.transport_layer.get_room_complexity(
+                destination=destination,
+                room_id=room_id
+            )
+            defer.returnValue(complexity)
+        except CodeMessageException:
+            # We didn't manage to get it -- probably a 404. We are okay if other
+            # servers don't give it to us.
+            pass
+        except Exception as e:
+            logger.exception(
+                "Failed to fetch room complexity via %s for %s: %s",
+                destination, room_id, str(e)
```

Comment:
does this not result in the exception message being logged twice (`str(e)` as well as `logger.exception`) ?

Reply:
this is still a thing

Reply:
oop, fixed